### PR TITLE
Add versionning on KB entries and their translations; fixes #1326

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

File: inc/knowbaseitem.class.php [hunk 1]
```diff
@@ -1939,6 +1942,14 @@ function getRights($interface='central') {
       return $values;
    }
 
+   function pre_updateInDB()
+   {
```
Comment (orthagh) [inline inc/knowbaseitem.class.php]: CS (open brace on the same line)

File: inc/knowbaseitem_revision.class.php [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+      $header .= "<th></th></tr>";
+      echo $header;
+
+      $user = new User();
+      $user->getFromDB($item->fields['users_id']);
+
+      //current contents
+      echo "<tr class='tab_bg_2'>";
+      echo "<td>(" . __('cur')  . ")</td>" .
+              "<td><input type=\"radio\" name=\"oldid\" value=\"0\" style=\"visibility:hidden\"/>" .
```
Comment (orthagh) [inline inc/knowbaseitem_revision.class.php]: ' instead of \\" should be better, no ?

**Review by orthagh — CHANGES_REQUESTED** [review]

File: inc/knowbaseitem_revision.class.php (hunk 2 above)
Comment (trasher) [reply inc/knowbaseitem_revision.class.php]: I had in mind HTML attributes must be quoted using double quotes, but after checking, seems I was wrong.

It was not well explained in old specs versions, but it is for HTML5.

Comment (orthagh) [reply inc/knowbaseitem_revision.class.php]: it's mandatory in html5 ?

Comment (trasher) [reply inc/knowbaseitem_revision.class.php]: Non, not at all. Using single ou double quotes has always been permitted, but the previous HTML specs were unclear ;)

**Review by orthagh — APPROVED** [review]

**Pull request merged**